### ISSUE TITLE
EnC: Changing signature of an async method or an iterator does not work

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#69199]
A state machine can be in-progress and emitting new kick-off method does not affect the old MoveNext method. The old method keeps executing but the execution no longer matches the code.

We should report a rude edit.

```C#
using System;
using System.Threading.Tasks;

class Program
{
    static async Task Main()
    {
        await F(1);
    }

    static async Task F(byte a)
    {
        Console.WriteLine($"1: {a}");
        await Task.Delay(1);
        Console.WriteLine($"2: {a}");
        await Task.Delay(1);
    }
}
```

After changing the parameter type to `int`, inserting some whitespace and stepping:

![image](https://github.com/dotnet/roslyn/assets/41759/88fe3ba6-c917-43f6-8e87-bed7ea227d29)
